clients/rust/tests/initialize_stake.rs

```diff
@@ -80,3 +71,287 @@ async fn initialize_config_with_mint_and_token() {
     assert_eq!(stake_account.validator, validator);
     assert_eq!(stake_account.authority, validator);
 }
+
+#[tokio::test]
+async fn fail_initialize_stake_with_initialized_account() {
+    let mut context = ProgramTest::new("stake_program", paladin_stake::ID, None)
+        .start_with_context()
+        .await;
+
+    // Given a config account and a validator's vote account.
+
+    let config = create_config(&mut context).await;
+    let validator = Pubkey::new_unique();
+    let vote = create_vote_account(&mut context, &validator, &validator);
+
+    // And we initialize the stake account.
+
+    let (stake_pda, _) = find_stake_pda(&validator, &config);
+
+    let transfer_ix = system_instruction::transfer(
+        &context.payer.pubkey(),
+        &stake_pda,
+        context
+            .banks_client
+            .get_rent()
+            .await
+            .unwrap()
+            .minimum_balance(Stake::LEN),
+    );
+
+    let initialize_ix = InitializeStakeBuilder::new()
+        .config(config)
+        .stake(stake_pda)
+        .validator_vote(vote)
+        .instruction();
+
+    let tx = Transaction::new_signed_with_payer(
+        &[transfer_ix, initialize_ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+    context.banks_client.process_transaction(tx).await.unwrap();
+
+    let account = get_account!(context, stake_pda);
+    assert_eq!(account.data.len(), Stake::LEN);
+
+    // When we try to initialize the stake account again.
+
+    let initialize_ix = InitializeStakeBuilder::new()
+        .config(config)
+        .stake(stake_pda)
+        .validator_vote(vote)
+        .instruction();
+
+    let tx = Transaction::new_signed_with_payer(
+        &[initialize_ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+    let err = context
+        .banks_client
+        .process_transaction(tx)
+        .await
+        .unwrap_err();
+
+    // Then we expect an error.
+
+    assert_instruction_error!(err, InstructionError::AccountAlreadyInitialized);
+}
+
+#[tokio::test]
+async fn fail_initialize_stake_without_funding_account() {
```

Comment:
Did you see this comment?
```
let account = context.banks_client.get_account(stake_pda).await.unwrap();
// TODO: Shouldn't we get an error from the instruction?
assert!(account.is_none());
```
The transaction succeeds but the account is not created. Is this because of how the tests are run using `banks_client`?

Reply:
Ahh I also noticed this in my tests! I have a note to check into it, but yeah I'm assuming it has to do with the way CPI is stubbed in `program-test`. Maybe drop this test for now?